### PR TITLE
More newbie-friendly documentation for the CodeNarc plugin regarding Groovy 3

### PR DESCRIPTION
**Pull request by JayWei1215**
More newbie-friendly documentation for the CodeNarc plugin regarding Groovy 3
Signed-off-by: Jay Wei <jaywei011215@gmail.com>

### Context

Provide documentation on specifying the Groovy version for the CodeNarc plugin dependency. 

Documentation change only.

Fixes https://github.com/gradle/gradle/issues/14853

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md).
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team.
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective.
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic.
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes.
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`.
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/documentation/docs/src/docs/userguide/core-plugins/codenarc_plugin.adoc [hunk 1]
```diff
@@ -86,6 +86,17 @@ The CodeNarc plugin adds the following dependency configurations:
 | The CodeNarc libraries to use
 |===
 
+[NOTE]
+====
+When there is a significant difference between the Groovy version used for compiling the code and the version that CodeNarc depends on, it's necessary to specify a specific Groovy version.
```
Comment (tresat) [inline platforms/documentation/docs/src/docs/userguide/core-plugins/codenarc_plugin.adoc:91]:
```suggestion
If CodeNarc requires a different Groovy version than that used to compile Groovy source,  you can supply one using the `codenarc` configuration.
```

**Review by tresat — CHANGES_REQUESTED** [review]

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by JayWei1215** [comment]
I have updated the PR @tresat

**Review by lkasso — APPROVED** [review]

**Comment by JayWei1215** [comment]
It still prompts '1 change requested' here, and I'm not sure if this is the reason for the merge branch failure. @lkasso @tresat

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by JayWei1215** [comment]
I have fixed the error. @lkasso @blindpirate

**Pull request merged**